utils/ctxutil/ctxutil.go

```diff
@@ -0,0 +1,68 @@
+package ctxutil
+
```

Comment:
code-wise this package is fine. Personally I prefer putting the 'ctx getter/setter' funcs in the package that defines the value that is transported via context.
This way you're mixing the tooling (ctx/ctxutils) with the business logic (default values). So another refactoring steps may be 
 - moving the getter/setter funcs
 - exposing the type assertion results `ok` (`value, ok := ctx.Value(key).(type)` as getter func return value.

Reply:
I didn't want to expose the type assertion results, as this makes using these method a little more annyoing, but you're right about mixing tooling with business logic. Will change that in a later PR. Right now the `ctxutil` is IMHO OK as it helped a lot in quickly introducing contexts.